### PR TITLE
feat: add ability to push expiration date to secret in azure key vault

### PR DESCRIPTION
**Pull request by deggja**
## Problem Statement

Currently, its not possible to push expiration date of a secret to Azure key vault using a PushSecret.

## Proposed Changes

Fetch expiration date from source secret annotation when pushing to Azure key vault to set it.

## Checklist

- [x] I have read the [contribution guidelines](https://external-secrets.io/latest/contributing/process/#submitting-a-pull-request)
- [x] All commits are signed with `git commit --signoff`
- [x] My changes have reasonable test coverage
- [x] All tests pass with `make test`
- [x] I ensured my PR is ready for review with `make reviewable`

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,6 +2,8 @@ module github.com/external-secrets/external-secrets
 
 go 1.23.1
 
+toolchain go1.23.3
```
Comment (Skarlso) [inline go.mod:5]: Please remove this.

Comment (deggja) [reply go.mod]: sorry, I removed it
I had some issues locally and did some clean up and this probably appeared because of it.

**Comment by gusfcarvalho** [comment]
hi @deggja ! thanks for your contribution.

In order to make this contribution in line with how other providers do it, you should allow it to be fetched via a pushSecret `metadata` block as opposed to annotations on the source Kubernetes Secret. See the design at https://github.com/external-secrets/external-secrets/pull/3612/files#diff-b20a64fd8b46ed86e84612368fa70c6e59805645112f5f35d943b925cc43f3beR88-R97  
Also - Kubernetes implementation for reference: https://github.com/external-secrets/external-secrets/blob/main/pkg/provider/kubernetes/metadata.go#L58

**Comment by deggja** [comment]
> hi @deggja ! thanks for your contribution.
> 
> 
> 
> In order to make this contribution in line with how other providers do it, you should allow it to be fetched via a pushSecret `metadata` block as opposed to annotations on the source Kubernetes Secret. I am particularly against having providers behaving differently, so if it would depend on me, I'd not consider this contribution to be merged as is 😅. 
> 
> 
> 
> See the design at https://github.com/external-secrets/external-secrets/pull/3612/files#diff-b20a64fd8b46ed86e84612368fa70c6e59805645112f5f35d943b925cc43f3beR88-R97  
> 
> 
> 
> Also - Kubernetes implementation for reference: https://github.com/external-secrets/external-secrets/blob/main/pkg/provider/kubernetes/metadata.go#L58 
> 
> 

Thats fair. I'll make the changes accordingly, thanks.

**Comment by Skarlso** [comment]
@gusfcarvalho We really should have provided a generic implementation here that people could use instead of copy and pasting it from Kubernetes. :)

**Comment by Skarlso** [comment]
The problem is that the spec part of this is arbitrary. But it's fine, we can make this work.

**Comment by Skarlso** [comment]
Okay, bare with me. I'm going to give you some code changes to make that thing generic from the Kubernetes thing so you can pass in your arbitrary push metadata spec. :)

**Comment by Skarlso** [comment]
Alternatively, we can merge this and I'll make the changes myself.

**Comment by Skarlso** [comment]
@deggja What would you like to do?

**Comment by Skarlso** [comment]
Another option is if I just push that code into your branch. :)

**Comment by Skarlso** [comment]
@deggja Okay, pushed the code, please review your changes if they are still okay. :) Thanks!

**Comment by Skarlso** [comment]
Ugh, I'll fix my linter issues. :D Sorry about that.

**Review by Skarlso — APPROVED** [review]

**Pull request merged**